internal/pkg/agent/application/upgrade/artifact/download/snapshot/downloader.go

```diff
@@ -66,8 +66,10 @@ func (e *Downloader) Reload(c *artifact.Config) error {
 
 // Download fetches the package from configured source.
 // Returns absolute path to downloaded package and an error.
-func (e *Downloader) Download(ctx context.Context, a artifact.Artifact, version string) (string, error) {
-	return e.downloader.Download(ctx, a, version)
+func (e *Downloader) Download(ctx context.Context, a artifact.Artifact, version *agtversion.ParsedSemVer) (string, error) {
+	// remove build metadata to match filename of the package for the specific snapshot build
+	strippedVersion := agtversion.NewParsedSemVer(version.Major(), version.Minor(), version.Patch(), version.Prerelease(), "")
```

Comment:
What happens if we don't do this? Why can't we have a version like `8.13.0-SNAPSHOT+buildYYYYMMDD` where -SNAPSHOT is the prerelease component?

Reply:
Right the problem is that we have decided that snapshot IDs are specified as build metadata, even though that is not actually the version in the artifacts API.  ability to handle the case where we actually want build metadata in the version.

We could solve this conflict in the artifacts API by doing something like:

https://snapshots.elastic.co/8.11.2-buildYYYYMMDD/downloads/beats/elastic-agent/elastic-agent-8.11.2-SNAPSHOT-darwin-aarch64.tar.gz

That said this doesn't exist yet, and because this is only for snapshots we can fix it without worrying about the stack release cadence. So let's just leave it as is.